pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -80,7 +87,24 @@ func (u UsernamePassword) Setup(baseDir string) (*url.URL, error) {
 	return overrideSourceURL, nil
 }
 
-func doSetup(sourceURL url.URL, usernameSecret, passwordSecret, tokenSecret string) (*url.URL, *url.URL, error) {
+type overrideURLFunc func(sourceURL *url.URL, username, password string) (*url.URL, error)
+
+func removeCredentials(sourceURL *url.URL, username, password string) (*url.URL, error) {
+	overrideURL := *sourceURL
+	overrideURL.User = nil
+	return &overrideURL, nil
+}
+
+func longPasswordOverride(dir string, overrideFn overrideURLFunc) overrideURLFunc {
+	return func(sourceURL *url.URL, username, password string) (*url.URL, error) {
+		if len(password) > curlPasswordThreshold {
+			return startProxy(dir, sourceURL, username, password)
```

Comment:
It means we'd only use the proxy for our own gitserver and leave the other paths as they are. In my mind, this introduces less risk because we don't change the way things have been working (and have been tested).